src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -14225,89 +14225,1275 @@ public static void Main()
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "C.M").WithArguments("System.Runtime.CompilerServices.DateTimeConstantAttribute", ".ctor").WithLocation(1, 9));
+        }
+
+        [Theory, WorkItem(65728, "https://github.com/dotnet/roslyn/issues/65728")]
+        [InlineData("DateTime ")]
+        [InlineData("DateTime?")]
+        public void MissingDateTimeConstantAttribute_ExternalMethodGroup_CustomType(string type)
+        {
+            var source1 = $$"""
+                using System;
+                using System.Runtime.CompilerServices;
+                using System.Runtime.InteropServices;
+
+                public delegate void Del([Optional, DateTimeConstant(100L)] {{type}} d);
+                
+                public class C
+                {
+                    public static void M([Optional, DateTimeConstant(100L)] {{type}} d) { }
+                }
+                """;
+            var comp1 = CreateCompilation(source1).VerifyDiagnostics();
+
+            var source2 = """
+                Del m = C.M;
+                """;
+            var comp2 = CreateCompilation(source2, new[] { comp1.ToMetadataReference() });
+            comp2.MakeTypeMissing(WellKnownType.System_Runtime_CompilerServices_DateTimeConstantAttribute);
+            comp2.VerifyEmitDiagnostics();
+        }
+
+        [Fact, WorkItem(65728, "https://github.com/dotnet/roslyn/issues/65728")]
+        public void MissingDateTimeConstantAttribute_OverloadResolution()
+        {
+            var source1 = """
+                using System;
+                using System.Runtime.CompilerServices;
+                using System.Runtime.InteropServices;
+
+                public delegate void Del([Optional, DateTimeConstant(100L)] DateTime d);
+
+                public class C
+                {
+                    public void M(Del d) { }
+                    public void M(Action<DateTime> d) { }
+                }
+                """;
+            var comp1 = CreateCompilation(source1).VerifyDiagnostics();
+
+            var source2 = """
+                using System;
+                using System.Runtime.CompilerServices;
+                using System.Runtime.InteropServices;
+
+                new C().M(([Optional, DateTimeConstant(100L)] DateTime d) => { });
+                """;
+
+            var diagnostics = new[]
+            {
+                // (5,9): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(Del)' and 'C.M(Action<DateTime>)'
+                // new C().M(([Optional, DateTimeConstant(100L)] DateTime d) => { });
+                Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(Del)", "C.M(System.Action<System.DateTime>)").WithLocation(5, 9)
+            };
+
+            var comp2 = CreateCompilation(source2, new[] { comp1.ToMetadataReference() });
+            comp2.MakeTypeMissing(WellKnownType.System_Runtime_CompilerServices_DateTimeConstantAttribute);
```

Comment:
We should also have some tests for when the constructor we're expecting is missing, but the type is present (use `MakeMemberMissing` instead).

Reply:
Thanks. Extended a few tests (not all combinations) to verify that the missing type error is reported as expected (and turns out I was reporting use-site diagnostics for the wrong constructor), plus these "overload resolution" ones.